### PR TITLE
[2.4.0 stack 13/18] Testing: fast-check properties, expectTypeOf signature pins and benchmarks in every test file

### PR DESCRIPTION
**Pull request by hyanmandian**
## What does this PR do?

Part 13 of 18 of the 2.4.0 release stack (main <- stack/01-tooling <- ... <- stack/11-testing). One commit: every util\x27s test file now ends with fast-check properties that hold by specification (generated values are valid, format/parse round-trip, masks never change the verdict, arbitrary input never throws, data tables agree with their validators), an `expectTypeOf` block that pins the public signature (checked statically by `vp check`) and, on hot paths, `bench` cases; `expectTypeOf` and `bench` come from the multi-runtime shim, so the same file runs on vitest, Bun and Deno.

## Commits in this part (1)

- 0525a5c test: property tests, signature pins and benchmarks in every test file

## How to review and merge

- Review each part on its own; the diff of this PR is exactly the commits above.
- Every part is green on its own: each branch builds, lints, passes the tests on Node 20/22/24/26, Bun, Deno and the four browsers, and passes the tree-shaking check against the part below it.
- Do not merge the lower parts individually. When all eighteen are approved, retarget #520 to `main` and merge it with a merge commit: the reviewed commits land unchanged and GitHub marks the seventeen parts below as merged.
- Zero breaking changes: every existing signature, export, entry point and error message is preserved. Twelve output corrections were validated against the published 2.3.0 tarball and are listed in `CHANGELOG.md` (part 7); the build of every later part is compared export by export against the part below it.

## Stack

- Previous: #522
- Next: #525
- Top of stack: #520

### CODE REVIEW

**Comment by hyanmandian** [comment]
@coderabbitai full review

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -76,7 +76,16 @@ example `formatSomething`):
    or `src/is-valid-service-phone/is-valid-service-phone.ts` for examples).
 3. Add tests alongside it in `src/format-something/format-something.test.ts`. Cover valid input,
    invalid/edge-case input, and options, if any. Tests must pass on Node, Bun and Deno (see
-   `npm test:bun` / `npm test:deno` under Useful scripts).
+   `npm test:bun` / `npm test:deno` under Useful scripts). Expectations are hand-written literals,
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply CONTRIBUTING.md:79]: Already fixed at the tip (#520, d0dba0a): CONTRIBUTING invokes every package script as npm run <script>, test:bun and test:deno included; a full-file sweep left only the plain npm commands (install, sbom, stage publish, stage approve, login, publish).

File: src/_internals/test/properties.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+ * @returns {void} Nothing.
+ */
+export const expectRoundTrip = <T>(
+	format: (value: T) => string,
+	parse: (value: string) => T,
+	arbitrary: fc.Arbitrary<T>,
+): void => {
+	fc.assert(
+		fc.property(arbitrary, (value) => {
+			expect(parse(format(value))).toBe(value);
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/_internals/test/properties.ts:176]: Fixed at the tip (#520, 4bbcbb5): every caller of expectRoundTrip passes primitives, so T is now constrained to number | string and toBe stays (by-value for primitives, and independent of the Bun/Deno shim's toEqual).

File: src/format-license-plate/format-license-plate.test.ts [hunk 3]
```diff
@@ -40,4 +44,44 @@ describe("formatLicensePlate", () => {
 		expect(formatLicensePlate("1ab4")).toBe("");
 		expect(formatLicensePlate("ab14")).toBe("");
 	});
+
+	describe("properties", () => {
+		test("should hyphenate an old format plate and leave a Mercosul one alone", () => {
+			fc.assert(
+				fc.property(fc.constantFrom("LLLNNNN", "LLLNLNN"), (format) => {
+					const plate = generateLicensePlate(format);
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/format-license-plate/format-license-plate.test.ts:52]: Fixed at the tip (#520, 4bbcbb5): a licensePlates(format) arbitrary was added to the shared arbitraries and the property no longer calls generateLicensePlate, so a failure replays from the fast-check seed.

File: src/generate-pix-payload/generate-pix-payload.test.ts [hunk 4]
```diff
@@ -401,4 +403,117 @@ describe("generatePixPayload", () => {
...
+
+		const txids = fc.stringMatching(/^[A-Za-z0-9]{1,25}$/);
+
+		const cents = fc.integer({ min: 1, max: 9_999_999 });
+
+		test("should round-trip a static payload through parsePixPayload", () => {
+			fc.assert(
+				fc.property(names, cities, (merchantName, merchantCity) => {
+					const key = generateCpf();
+					const payload = generatePixPayload({ key, merchantName, merchantCity });
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/generate-pix-payload/generate-pix-payload.test.ts:422]: Fixed at the tip (#520, 4bbcbb5): the properties use fixed valid CPF and CNPJ constants; the non-property round-trip loops that intentionally exercise the generators keep them.

File: src/generate-processo-juridico/generate-processo-juridico.test.ts [hunk 5]
```diff
@@ -77,4 +82,62 @@ describe("generateProcessoJuridico", () => {
 			Math.random = originalRandom;
 		}
 	});
+
+	describe("properties", () => {
+		const year = fc.integer({ min: currentYear, max: 9999 });
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/generate-processo-juridico/generate-processo-juridico.test.ts:87]: Fixed at the tip (#520, 4bbcbb5): the year is read at run time inside the property (fc.pre against the current year) and the cast is gone, replaced by runtime not-null/length checks. The runtime shim has no fake timers, so the clock is read per call instead of frozen.

File: src/get-states/get-states.test.ts [hunk 6]
```diff
@@ -64,4 +71,37 @@ describe("getStates", () => {
...
+					expect(getStateByIbgeCode(state.ibgeCode)).toEqual(state);
+				}),
+			);
+		});
+	});
+});
+
+describe("getStates types", () => {
+	test("should take no arguments and return an array of State", () => {
+		expectTypeOf(getStates).parameter(0).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/get-states/get-states.test.ts:97]: Fixed at the tip (#520, 4bbcbb5): expectTypeOf(getStates).parameters.toEqualTypeOf<[]>(); that was the only parameter(0).toBeUndefined() in the tree.

File: src/is-valid-bank-account/is-valid-bank-account.test.ts [hunk 7]
```diff
@@ -1540,4 +1550,95 @@ describe("isValidBankAccount", () => {
...
+		});
+
+		test("should accept any single digit for a bank validated by structure only", () => {
+			const structureOnly = fc.constantFrom("077", "085", "197", "290", "336", "748", "756");
+
+			fc.assert(
+				fc.property(structureOnly, agencies, accounts, (bankCode, agency, pool) => {
+					const account = pool.slice(0, 6);
+
+					expect(isValidBankAccount({ bankCode, agency, account, digit: "7" })).toBe(true);
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/is-valid-bank-account/is-valid-bank-account.test.ts:1594]: Fixed at the tip (#520, 4bbcbb5): the property draws the check digit from fc.integer({ min: 0, max: 9 }) and asserts with it.

File: src/is-valid-iban/is-valid-iban.test.ts [hunk 8]
```diff
@@ -1,6 +1,13 @@
-import { describe, expect, test } from "../_internals/test/runtime";
+import * as fc from "fast-check";
+
+import { describe, expect, expectTypeOf, test } from "../_internals/test/runtime";
 import { isValidIban } from "./is-valid-iban";
 
+const CHECK_DIGITS = Array.from({ length: 97 }, (_, index) => String(index).padStart(2, "0"));
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/is-valid-iban/is-valid-iban.test.ts:6]: Fixed at the tip (#520, 4bbcbb5): CHECK_DIGITS now spans 02 to 98, the ISO 7064 MOD 97-10 range; for the body 00000000000000000000066CA the accepted pair is 98 (the old 00-96 range silently returned the non-canonical 01). The 97 values cover every residue once, so the exactly-one-pair property is sound.

File: src/parse-iban/parse-iban.test.ts [hunk 9]
```diff
@@ -1,7 +1,19 @@
...
+import * as fc from "fast-check";
+
+import { describe, expect, expectTypeOf, test } from "../_internals/test/runtime";
 import { formatIban } from "../format-iban/format-iban";
 import { isValidIban } from "../is-valid-iban/is-valid-iban";
-import { parseIban } from "./parse-iban";
+import { parseIban, type Iban } from "./parse-iban";
+
+const findBrazilianIban = (body: string): string => {
+	for (let pair = 0; pair < 97; pair++) {
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply src/parse-iban/parse-iban.test.ts:9]: Fixed at the tip (#520, 4bbcbb5): findBrazilianIban loops 02 to 98. The library keeps its mod-97 routine private to isValidIban, so the tests keep searching instead of duplicating the algorithm.

File: vite.config.ts [hunk 10]
```diff
@@ -448,6 +448,10 @@ export default defineConfig({
 	},
 	test: {
 		exclude: ["**/node_modules/**", "**/dist/**", "**/.stryker-tmp/**", "**/reports/**"],
+		benchmark: {
+			include: ["src/**/*.test.ts"],
```
[the remark this answers is not in the document]
Comment (hyanmandian) [reply vite.config.ts:452]: Not changing this. Benches live in the same .test.ts file on purpose (the runtime shim registers bench as test.todo outside benchmark mode so the file runs on vitest, Bun and Deno), and the failure does not reproduce: npx vp test bench --run over the whole repository and over src/parse-boleto/parse-boleto.test.ts both complete with exit code 0, six benchmark suites reported, all in files that also declare test/it.
